### PR TITLE
[SUP-592] Distinguish no billing account from no access to billing account

### PR DESCRIPTION
**Pull request by nawatts**
When viewing a billing project, currently "Billing Account: No Access" is displayed in two different cases:
- When the user does not have access to the billing project's billing account
- When the billing project has no billing account associated with it

This distinguishes those two cases by changing the text to either "No access to linked billing account" or "No linked billing account".

The path for the billing page is `#billing`. Or it can be reached from the main sidebar (expand the user menu).

### CODE REVIEW

File: src/pages/billing/Project.js
Comment (nawatts) [inline src/pages/billing/Project.js:535]: I don't know if it's possible that a billing account has no display name, but this handles it in case it is.

File: src/pages/billing/Project.js
Comment (nawatts) [inline src/pages/billing/Project.js:529]: This condition is taken from existing code that controls the disabled state of the "Remove Billing Account" button.

**Comment by cahrens** [comment]
@rtitle's open PR addresses this issue, so I think you should not merge this change. https://github.com/DataBiosphere/terra-ui/pull/3117/files#diff-63f09855608a6a241dc6a5b78476307d1ea6ef749c662b0bf3323c2fc87a8f28R654

**Comment by nawatts** [comment]
> open PR addresses this issue, so I think you should not merge this change

Is Terra not having access to billing scopes the only reason that this returns undefined?
```js
_.find({ accountName: billingProject.billingAccount }, billingAccounts)
```

**Review by kyuksel — APPROVED** [review]
- Code LGTM and I verified the `No access to linked billing account` case. I don't have a project set up for the `No linked billing account` case.

- Approving the functionality…pending [Christina's comment](https://github.com/DataBiosphere/terra-ui/pull/3123#issuecomment-1151476006) is resolved.

**Comment by cahrens** [comment]
@nawatts I don't know the answer to your question, but the behavior is different with Rob's change and I think it is sufficient to differentiate between the 2 cases.

Try it out and see what you think.

**Comment by nawatts** [comment]
Google won't let me grant billing scopes to dev Terra, so I'm not sure how to try it out.

**Comment by cahrens** [comment]
@nawatts did you try logging in as Hermione?

**Comment by nawatts** [comment]
Using Hermione on dev Terra, I can see billing accounts. With #3117, I get the "View billing account" button. Clicking that button opens the Google prompt to allow Terra to access billing.

**Comment by cahrens** [comment]
> Using Hermione on dev Terra, I can see billing accounts. With #3117, I get the "View billing account" button. Clicking that button opens the Google prompt to allow Terra to access billing.

Right, so in the PR "View Billing Account" is the equivalent of "No access". If you think there is some further improvements to be made, I would recommend that you wait until @rtitle's PR merges so that they play nicely together.

**Comment by nawatts** [comment]
> Right, so in the PR "View Billing Account" is the equivalent of "No access".

I think that's sometimes, but not always, true. This discussion has led me to realize that "No access" currently means one of _three_ things, not just the two noted in [SUP-592](https://broadworkbench.atlassian.net/browse/SUP-592). Currently, "No access" is displayed when:
1. Terra does not have access to the billing scope
2. The user has no access to the billing project's billing account
3. The billing project has no billing account

It sounds like #3117 resolves the first one, but I think there will still be a need for this PR to resolve the ambiguity between 2 and 3.

In any case though, point taken about making sure that the two work together. I moved this PR to draft until after #3117 merges.

**Review by dmohs — APPROVED** [review]

**Comment by rtitle** [comment]
Sounds good -- FWIW I iterated with Lou a bit on the approach in #3117

**Comment by nawatts** [comment]
> Sounds good -- FWIW I iterated with Lou a bit on the approach in https://github.com/DataBiosphere/terra-ui/pull/3117

I definitely like the addition of the "View billing account" button. It solves an issue that I didn't even realize was there. Currently, it seems like the only way to prompt to add the billing scope is to click the button to create a billing project.

**Comment by nawatts** [comment]
Rebased to resolve merge conflicts. After trying this out now that #3117 has merged, I still think this change is relevant. Currently, billing projects with no billing account say "No access", which is not accurate.

**Pull request merged**
